### ISSUE TITLE
Module related commands need to be refreshed for more Composer goodness

### ISSUE THREAD

**Issue by mondrake** [hechoendrupal/drupal-console#4180]
The module related commands need some refactoring, since Composer usage now is much more mature in Drupal.

For example:
module:download - it does not work; it does not have an option to download via composer
module:install - updated in #4167 , but still falling back to a non-working tarball download if --composer option is not specified
module:update - only works with --composer, but it should be refreshed after #4167 
module:uninstall - should check if it can be removed via composer after drupal-uninstalling

More?

**Comment by enzolutions** [comment]
@mondrake you want to take care of these tasks?

**Comment by mondrake** [comment]
Yes. If anyone has further input on what to do in this space, feel free to comment.

**Comment by mondrake** [comment]
@enzolutions can you please review https://github.com/hechoendrupal/drupal-console-core/pull/378 and https://github.com/hechoendrupal/drupal-console-core/pull/377 that are preliminary PRs before being able to address this.

**Comment by mondrake** [comment]
@enzolutions before I submit a PR for this, please see some considerations.

I believe downloading modules as tarballs is a dead story. We should focus on the new Composer way instead, and try to integrate the 'composer' tasks of downloading modules in the codebase with the 'drupal' tasks of installing, uninstalling and updating modules.

Following this logic, I suggest to completely remove the tarball downloading code, and reorganise the module commands as follows:
- `module:download` this is mostly equivalent to `composer require`, it's actually performing this as a subprocess. The goodness here is that you can mix modules names in 'drupal' and 'composer' syntax and that trying to composer-require modules that are part of Drupal core will be ignored.
This Drupal Console command `drupal module:download pagerer drupal/token file_mdm_exif:^2` will lead to the execution of `composer require drupal/pagerer drupal/token drupal/file_mdm_exif:^2`.
However, `module:download` may also not be used, since `module:install` and `module:update` can perform all the downloading, installing and updating tasks in a single command.
  
- `module:install` is Drupal-installing the modules specified, and additionally is composer-requiring them in advance if they are not available in the codebase.

- `module:update` is composer-updating the modules specified, or composer-requiring them if a constraint is given (for example, to update between different major versions of a module). 
Additionally, it will run `update:execute` at the end of the composer process (unless specified not to), so that with a single Drupal Console command you can perform update of the codebase and of the schema/settings of a module.

I will post a PR for this; `module:uninstall` will be left out for now and follow with another PR.
